src/remote_prune

```diff
@@ -0,0 +1,236 @@
+        # also be present, and the duration of cloud-uploads must be equal or shorter
+        if "build" in policy[stream].keys():
+            cloudUploadsCheck(policy[stream])
+        buildJsonData = getBuildsForStream(stream)
+        # action is basically whatever is needed to be pruned for the respective stream
+        for action in policy[stream]:
+            years = policy[stream][action]
+            #Ref date: Artifacts older than this date will be deleted
+            refDate = datetime.datetime.now() - relativedelta(years=int(years[:-1]))
+            print(f"Reference Date for action {action} is {refDate}")
+
+            for index, build in enumerate(buildJsonData["builds"]):
+                build_id = build["id"]
+                if "policy-cleanup" in build.keys():
+                    if action in build["policy-cleanup"]:
+                        print(f"The {build_id} has already had {action} pruning completed")
+                        break
+                timestamp = build_id.split('.')[1]
+                buildDate = datetime.datetime(int(timestamp[0:4]), int(timestamp[4:6]), int(timestamp[-2:]))
+                if buildDate < refDate:
+                    # For testing pusposes only using one build
+                    if build_id == "35.20210508.91.0":
+                        for arch in build["arches"]:
+                            print(f"Pruning {years} old {action} for {build_id} for {arch} in {stream}")
+                            # buildFetch(args.stream, build_id, arch)
+                            metaJson = open(f"builds/{build_id}/{arch}/meta.json")
+                            metaData = json.load(metaJson)
+
+                            images = {
+                                "amis": metaData.get("amis") or [],
+                                "azure": metaData.get("azure") or [],
+                                "gcp": metaData.get("gcp") or [],
+                            }
+                            currentBuild = Build(
+                                id=build_id,
+                                timestamp=timestamp,
+                                images=images,
+                                arches=arch,
+                            )
+                            match action:
+                                case "cloud-uploads":
+                                    delete_cloud_resource(currentBuild, cloud_config, args.dry_run)
+                                    if not args.dry_run:
+                                        build.setdefault("policy-cleanup", []).append(action)
+                                        buildJsonData["builds"][index] = build
+                                case "build":
+                                    print(f"Will delete everything for {build_id}")
+        # pprint.pprint(buildJsonData['builds'], compact=True)
+        with open(f"builds/builds.json", "w") as json_file:
+            json_file.write(json.dumps(buildJsonData))
```

Comment:
just pointing out we will need some special handling of builds.json when we upload it back to s3